xonsh/jobs.py

```diff
@@ -251,9 +253,15 @@ def fg(args, stdin=None):
         act = tasks[0]  # take the last manipulated task by default
     elif len(args) == 1:
         try:
-            act = int(args[0])
-        except ValueError:
+            if args[0] == '+': # take the last manipulated task
+                 act = tasks[0]
+            elif args[0] == '-': # take the second to last manipulated task
+                act = tasks[1]
+            else:
+                act = int(args[0])
+        except (ValueError, IndexError):
             return '', 'Invalid job: {}\n'.format(args[0])
```

Comment:
I don't think you're missing anything. Maybe I should re-write it in a cleaner way? That method could easily get out of hand.